wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -203,4 +205,27 @@ protected OAuthAuthenticator getAuthenticator(String oauthProviderName) throws B
         }
         return oauth;
     }
+
+    @POST
+    @Path("token")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public void setToken(@Required @QueryParam("oauth_provider") String oauthProvider, //
```

Comment:
Looks like one of  map's entry can growth infinitely because of this API, and it is not restricted so any user of the product can cause OOM